evm/interchainTokenFactory.js

```diff
@@ -207,6 +209,43 @@ async function processCommand(config, chain, options) {
             break;
         }
 
+        case 'registerGatewayTokens': {
+            let { assetApi, tokenInfoApi, batchSize } = options;
+
+            assetApi = assetApi || `${config.axelar.lcd}/axelar/nexus/v1beta1/assets/`;
+            tokenInfoApi = tokenInfoApi || `${config.axelar.lcd}/axelar/evm/v1beta1/token_info/`;
+            batchSize = batchSize || 10;
+
+            validateParameters({ isString: { assetApi, tokenInfoApi }, isNumber: { batchSize } });
+
+            const { assets } = (await axios.get(`${assetApi}${chain.name}`)).data;
+
+            for (let i = 0; i < assets.length; i += batchSize) {
+                const asset = assets[i];
+                const salt = keccak256(hexlify(toUtf8Bytes(asset)));
+                const tokenId = await interchainTokenService.interchainTokenId(AddressZero, salt);
+                const tokenManagerAddress = await interchainTokenService.tokenManagerAddress(tokenId);
+
+                if ((await provider.getCode(tokenManagerAddress)).length > 2) {
```

Comment:
the logic is fragile to changes in the batch size variable. split this into two separate loops. one that collects the list of all assets that aren't registered, and another that batch registers them based on the batch size. You can make it more functional by using map/filter.